cherrypy/_cpreqbody.py

```diff
@@ -724,14 +738,22 @@ def default_proc(self):
                 self.file = result
 
     def read_into_file(self, fp_out=None):
-        """Read the request body into fp_out (or make_file() if None).
+        """Read the request body into fp_out.
+
+        if fp_out not given, use either make_file or
+        a BytesIO object, based on maxrambytes.
 
         Return fp_out.
         """
-        if fp_out is None:
-            fp_out = self.make_file()
-        self.read_lines_to_boundary(fp_out=fp_out)
-        return fp_out
+        if not fp_out:
+            x = self.read_lines_to_boundary()
+            if isinstance(x,(str,bytes)):
```

Comment:
I'm not sure whether this block would work under both Python 2 and Python 3. There's [`_cpcompat` module having helper functions](https://github.com/cherrypy/cherrypy/blob/1639205/cherrypy/_cpcompat.py#L27-L91) for dealing with different string/byte representations.